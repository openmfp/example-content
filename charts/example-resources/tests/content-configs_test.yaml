suite: content-configurations
templates:
  - content-configurations/*.yaml
tests:
  - it: matches the snapshot
    asserts:
      - matchSnapshot: {}
