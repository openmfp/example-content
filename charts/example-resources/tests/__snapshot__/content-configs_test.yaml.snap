matches the snapshot:
  1: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: home
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: "\n{\n  \"name\": \"overview\",\n  \"creationTimestamp\": \"2022-05-17T11:37:17Z\",\n  \"luigiConfigFragment\": [\n    {\n      \"data\": {\n        \"nodes\": [\n          {\n            \"entityType\": \"global\",\n            \"pathSegment\": \"home\",\n            \"hideFromNav\": true,\n            \"defineEntity\": {\n              \"id\": \"main\"\n            },\n            \"children\": [\n              {\n                \"pathSegment\": \"overview\",\n                \"label\": \"Overview\",\n                \"icon\": \"home\",\n                \"defineEntity\": {\n                  \"id\": \"overview\"\n                },\n                \"compound\": {\n                  \"renderer\" : {\n                    \"use\": \"grid\",\n                    \"config\": {\n                      \"columns\": \"1fr 1fr 1fr 1fr\"\n                    }\n                  }\n                }\n              }\n            ]\n          },{\n            \"entityType\": \"main.overview::compound\",\n            \"url\": \"https://luigiwebcomponents.gitlab.io/layouts/panelHeader.js\",\n            \"context\": {\n              \"title\": \"Welcome to OpenMFP Portal\",\n              \"description\": \" \"\n            },\n            \"layoutConfig\": {\n              \"row\": \"1\",\n              \"column\": \"1 / -1\"\n            }\n          },\n          \n          \n          \n          {\n            \"entityType\": \"main.account\",\n            \"pathSegment\": \"dashboard\",\n            \"label\": \"Dashboard\",\n            \"defineEntity\": {\n              \"id\": \"dashboard\"\n            },\n            \"compound\": {\n              \"renderer\" : {\n                \"use\": \"grid\",\n                \"config\": {\n                  \"columns\": \"1fr 1fr 1fr 1fr\"\n                }\n              }\n            }\n          },{\n            \"entityType\": \"main.account.dashboard::compound\",\n            \"url\": \"https://luigiwebcomponents.gitlab.io/layouts/panelHeader.js\",\n            \"context\": {\n              \"title\": \"Account Level Dashboard\",\n              \"description\": \"Comprehensive account overview dashboard for an account running inside an ApeiroRA instance\"\n            },\n            \"layoutConfig\": {\n              \"row\": \"1\",\n              \"column\": \"1 / -1\"\n            }\n          }\n          \n          \n        ]\n      }\n    }\n  ]\n}"
        contentType: json
matches the snapshot (gardener):
  1: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: gardener
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: |2-
          {
            "name": "gardener",
            "creationTimestamp": "2022-05-17T11:37:17Z",
            "luigiConfigFragment": [
              {
                "data": {
                  "nodes": [
                    {
                      "entityType": "main.account",
                      "pathSegment": "gardener",
                      "label": "Projects",
                      "category": {
                        "label": "Gardener",
                        "icon": "https://d.ing.gardener-op.mfp-dev.shoot.canary.k8s-hana.ondemand.com/static/assets/logo.svg",
                        "collapsible": false
                      },
                      "virtualTree": true,
                      "initialRoute": "namespace",
                      "url": "https://d.ing.gardener-op.mfp-dev.shoot.canary.k8s-hana.ondemand.com",
                      "icon": "https://d.ing.gardener-op.mfp-dev.shoot.canary.k8s-hana.ondemand.com/static/assets/logo.svg",
                      "loadingIndicator": {
                        "enabled": false
                      },
                      "requiredIFramePermissions": {
                        "sandbox": [
                          "allow-forms",
                          "allow-modals",
                          "allow-popups",
                          "allow-popups-to-escape-sandbox",
                          "allow-same-origin",
                          "allow-scripts"
                        ],
                        "allow": [
                          "clipboard-read",
                          "clipboard-write"
                        ]
                      }
                    }
                  ]
                }
              }
            ]
          }
        contentType: json
  2: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: home
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: "\n{\n  \"name\": \"overview\",\n  \"creationTimestamp\": \"2022-05-17T11:37:17Z\",\n  \"luigiConfigFragment\": [\n    {\n      \"data\": {\n        \"nodes\": [\n          {\n            \"entityType\": \"global\",\n            \"pathSegment\": \"home\",\n            \"hideFromNav\": true,\n            \"defineEntity\": {\n              \"id\": \"main\"\n            },\n            \"children\": [\n              {\n                \"pathSegment\": \"overview\",\n                \"label\": \"Overview\",\n                \"icon\": \"home\",\n                \"defineEntity\": {\n                  \"id\": \"overview\"\n                },\n                \"compound\": {\n                  \"renderer\" : {\n                    \"use\": \"grid\",\n                    \"config\": {\n                      \"columns\": \"1fr 1fr 1fr 1fr\"\n                    }\n                  }\n                }\n              }\n            ]\n          },{\n            \"entityType\": \"main.overview::compound\",\n            \"url\": \"https://luigiwebcomponents.gitlab.io/layouts/panelHeader.js\",\n            \"context\": {\n              \"title\": \"Welcome to OpenMFP Portal\",\n              \"description\": \" \"\n            },\n            \"layoutConfig\": {\n              \"row\": \"1\",\n              \"column\": \"1 / -1\"\n            }\n          },\n          \n          \n          \n          {\n            \"entityType\": \"main.account\",\n            \"pathSegment\": \"dashboard\",\n            \"label\": \"Dashboard\",\n            \"defineEntity\": {\n              \"id\": \"dashboard\"\n            },\n            \"compound\": {\n              \"renderer\" : {\n                \"use\": \"grid\",\n                \"config\": {\n                  \"columns\": \"1fr 1fr 1fr 1fr\"\n                }\n              }\n            }\n          },{\n            \"entityType\": \"main.account.dashboard::compound\",\n            \"url\": \"https://luigiwebcomponents.gitlab.io/layouts/panelHeader.js\",\n            \"context\": {\n              \"title\": \"Account Level Dashboard\",\n              \"description\": \"Comprehensive account overview dashboard for an account running inside an ApeiroRA instance\"\n            },\n            \"layoutConfig\": {\n              \"row\": \"1\",\n              \"column\": \"1 / -1\"\n            }\n          }\n          \n          \n        ]\n      }\n    }\n  ]\n}"
        contentType: json
matches the snapshot (infra):
  1: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: home
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: "\n{\n  \"name\": \"overview\",\n  \"creationTimestamp\": \"2022-05-17T11:37:17Z\",\n  \"luigiConfigFragment\": [\n    {\n      \"data\": {\n        \"nodes\": [\n          {\n            \"entityType\": \"global\",\n            \"pathSegment\": \"home\",\n            \"hideFromNav\": true,\n            \"defineEntity\": {\n              \"id\": \"main\"\n            },\n            \"children\": [\n              {\n                \"pathSegment\": \"overview\",\n                \"label\": \"Overview\",\n                \"icon\": \"home\",\n                \"defineEntity\": {\n                  \"id\": \"overview\"\n                },\n                \"compound\": {\n                  \"renderer\" : {\n                    \"use\": \"grid\",\n                    \"config\": {\n                      \"columns\": \"1fr 1fr 1fr 1fr\"\n                    }\n                  }\n                }\n              }\n            ]\n          },{\n            \"entityType\": \"main.overview::compound\",\n            \"url\": \"https://luigiwebcomponents.gitlab.io/layouts/panelHeader.js\",\n            \"context\": {\n              \"title\": \"Welcome to OpenMFP Portal\",\n              \"description\": \" \"\n            },\n            \"layoutConfig\": {\n              \"row\": \"1\",\n              \"column\": \"1 / -1\"\n            }\n          },\n          \n          \n          \n          {\n            \"entityType\": \"main.account\",\n            \"pathSegment\": \"dashboard\",\n            \"label\": \"Dashboard\",\n            \"defineEntity\": {\n              \"id\": \"dashboard\"\n            },\n            \"compound\": {\n              \"renderer\" : {\n                \"use\": \"grid\",\n                \"config\": {\n                  \"columns\": \"1fr 1fr 1fr 1fr\"\n                }\n              }\n            }\n          },{\n            \"entityType\": \"main.account.dashboard::compound\",\n            \"url\": \"https://luigiwebcomponents.gitlab.io/layouts/panelHeader.js\",\n            \"context\": {\n              \"title\": \"Account Level Dashboard\",\n              \"description\": \"Comprehensive account overview dashboard for an account running inside an ApeiroRA instance\"\n            },\n            \"layoutConfig\": {\n              \"row\": \"1\",\n              \"column\": \"1 / -1\"\n            }\n          }\n          \n          \n        ]\n      }\n    }\n  ]\n}"
        contentType: json
  2: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: infra
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: |2-
          {
            "name": "overview",
            "creationTimestamp": "2022-05-17T11:37:17Z",
            "luigiConfigFragment": [
              {
                "data": {
                  "nodes": [
                    {
                      "pathSegment": "ironcore",
                      "label": "Compute",
                      "entityType": "main.account",
                      "category": {
                        "label": "Iron Core",
                        "icon": "https://avatars.githubusercontent.com/u/147836484?s=200&v=4",
                        "collapsible": false
                      },
                      "loadingIndicator": {
                        "enabled": false
                      },
                      "keepSelectedForChildren": true,
                      "url": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/example-content/index.html#/iron-core/compute",
                      "children": [ ],
                      "context": {
                        "title": "Iron Core",
                        "content": " "
                      }
                    },
                    {
                      "pathSegment": "network",
                      "label": "Network",
                      "entityType": "main.account",
                      "category": "Iron Core",
                      "loadingIndicator": {
                        "enabled": false
                      },
                      "keepSelectedForChildren": true,
                      "url": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/example-content/index.html#/iron-core/network",
                      "children": [ ],
                      "context": {
                        "title": "OpenMCP",
                        "content": " "
                      }
                    },
                    {
                      "pathSegment": "openmcp",
                      "label": "Control Planes",
                      "entityType": "main.account",
                      "category": {
                        "label": "OpenMCP",
                        "icon": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/wc/assets/openmcp.png",
                        "collapsible": false
                      },
                      "loadingIndicator": {
                        "enabled": false
                      },
                      "keepSelectedForChildren": true,
                      "url": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/example-content/index.html#/openmcp/control-planes",
                      "children": [ ],
                      "context": {
                        "title": "OpenMCP",
                        "content": " "
                      }
                    }
                  ]
                }
              }
            ]
          }
        contentType: json
matches the snapshot (providers):
  1: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: home
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: "\n{\n  \"name\": \"overview\",\n  \"creationTimestamp\": \"2022-05-17T11:37:17Z\",\n  \"luigiConfigFragment\": [\n    {\n      \"data\": {\n        \"nodes\": [\n          {\n            \"entityType\": \"global\",\n            \"pathSegment\": \"home\",\n            \"hideFromNav\": true,\n            \"defineEntity\": {\n              \"id\": \"main\"\n            },\n            \"children\": [\n              {\n                \"pathSegment\": \"overview\",\n                \"label\": \"Overview\",\n                \"icon\": \"home\",\n                \"defineEntity\": {\n                  \"id\": \"overview\"\n                },\n                \"compound\": {\n                  \"renderer\" : {\n                    \"use\": \"grid\",\n                    \"config\": {\n                      \"columns\": \"1fr 1fr 1fr 1fr\"\n                    }\n                  }\n                }\n              }\n            ]\n          },{\n            \"entityType\": \"main.overview::compound\",\n            \"url\": \"https://luigiwebcomponents.gitlab.io/layouts/panelHeader.js\",\n            \"context\": {\n              \"title\": \"Welcome to OpenMFP Portal\",\n              \"description\": \" \"\n            },\n            \"layoutConfig\": {\n              \"row\": \"1\",\n              \"column\": \"1 / -1\"\n            }\n          },\n          \n          \n          \n          {\n            \"entityType\": \"main.account\",\n            \"pathSegment\": \"dashboard\",\n            \"label\": \"Dashboard\",\n            \"defineEntity\": {\n              \"id\": \"dashboard\"\n            },\n            \"compound\": {\n              \"renderer\" : {\n                \"use\": \"grid\",\n                \"config\": {\n                  \"columns\": \"1fr 1fr 1fr 1fr\"\n                }\n              }\n            }\n          },{\n            \"entityType\": \"main.account.dashboard::compound\",\n            \"url\": \"https://luigiwebcomponents.gitlab.io/layouts/panelHeader.js\",\n            \"context\": {\n              \"title\": \"Account Level Dashboard\",\n              \"description\": \"Comprehensive account overview dashboard for an account running inside an ApeiroRA instance\"\n            },\n            \"layoutConfig\": {\n              \"row\": \"1\",\n              \"column\": \"1 / -1\"\n            }\n          }\n          \n          \n        ]\n      }\n    }\n  ]\n}"
        contentType: json
  2: |
    apiVersion: core.openmfp.io/v1alpha1
    kind: ContentConfiguration
    metadata:
      name: providers
      namespace: NAMESPACE
    spec:
      inlineConfiguration:
        content: |2-
          {
            "name": "overview",
            "creationTimestamp": "2022-05-17T11:37:17Z",
            "luigiConfigFragment": [
              {
                "data": {
                  "nodes": [
                    {
                      "pathSegment": "marketplace",
                      "label": "Marketplace",
                      "entityType": "main",
                      "loadingIndicator": {
                        "enabled": false
                      },
                      "keepSelectedForChildren": true,
                      "url": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/example-content/index.html#/global-catalog",
                      "children": [{
                          "pathSegment": ":id",
                          "url": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/example-content/index.html#/catalog-item/:id",
                          "hideFromNav": true,
                          "label": "Service Details"
                      }]
                    },
                    {
                      "pathSegment": "capabilities",
                      "label": "Capabilities",
                      "entityType": "main.account",
                      "loadingIndicator": {
                        "enabled": false
                      },
                      "keepSelectedForChildren": true,
                      "url": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/example-content/index.html#/global-catalog",
                      "children": [{
                          "pathSegment": ":id",
                          "url": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/example-content/index.html#/catalog-item/:id",
                          "hideFromNav": true,
                          "label": "Service Details"
                      }]
                    },
                    {
                      "entityType": "main.overview::compound",
                      "url": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/wc/main.js#catalog",
                      "context": {
                        "title": "Enabled services"
                      },
                      "layoutConfig": {
                        "row": "auto",
                        "column": "1 / -1"
                      },
                      "webcomponent": {
                        "selfRegistered": true
                      }
                    },
                    {
                      "entityType": "main.account.dashboard::compound",
                      "url": "https://content.d1.openmfp.dxp.k8s.ondemand.com/ui/wc/main.js#catalog",
                      "context": {
                        "title": "Resource Overview"
                      },
                      "layoutConfig": {
                        "row": "auto",
                        "column": "1 / -1"
                      },
                      "webcomponent": {
                        "selfRegistered": true
                      }
                    }
                  ]
                }
              }
            ]
          }
        contentType: json
